internal/c/libqb.cpp

```diff
@@ -37148,6 +37148,26 @@ int32 func__resize() {
     return 0;
 }
 
+void sub__windowscale(int32 amount) {
+    int32 tempx, tempy;
+        if (amount < 50) return; // don't scale the screen to less than 50% of it's original size!!
+
+        tempx = environment_2d__screen_width * amount / 100;  // basically our width of the screen
+        tempy = environment_2d__screen_height * amount / 100; // and this is the height of the screen
+        if (tempx < 320) return; // minimal window width of 320.
+        if (tempy < 240) return; // minimal window height of 240
+        
+        ScreenResize = 1;
+        ScreenResizeScale = amount;

```

Comment:
I'm not sure how `ScreenResizeScale` works, you set it to the `amount` to scale, but the only usage of that variable is `resize_auto = ScreenResizeScale;`, and `resize_auto` is just supposed to be either `0`, `1` or `2` to indicate stretch or smooth scaling.

Looking at it closer, I think the thing missing is how we tell the GLUT thread to resize to a set window size. It's hard to work out how to do it with only the existing logic, but it seems like it wouldn't be too hard to add something around where `resize_auto` is handled that just forces the window to a size we supply from here, and then the rest of the logic after that should just work out.

Reply:
This is a magically command/variable.  It must work somewhere in the core files, because it handles the resizing perfectly, whereas just setting resize_auto itself doesn't.  With resize_auto, you have to calculate those new window limits and store them yourself.  With ScreenResizeScale, it's handled automatically somewhere.